utils/src/newtype.rs

```diff
@@ -0,0 +1,31 @@
+#[macro_export]
+macro_rules! newtype {
+    ($vis:vis $name:ident($wrapped:ty)) => {
```

Comment:
I wonder if that would be more idiomatic to more closely follow the usual Rust syntax inside the macro. I.e. the syntax to match would be `$vis struct $name($wrapped)` (notice the struct keyword). Maybe even consider a (optional?) semicolon at the end.

Reply:
Btw does the `$vis:vis` pattern match empty (i.e. private) visibility?